nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1364,12 +1367,16 @@ public Description matchClass(ClassTree tree, VisitorState state) {
       EnclosingEnvironmentNullness.instance(state.context).clear();
     } else {
       // handle the case where the top-class is unannotated, but there is a @NullMarked annotation
-      // on a nested class
-      // TODO handle @NullUnmarked once it is finalized
+      // on a nested class, or, conversely the top-level is annotated but there is a @NullUnmarked
+      // annotation on a nested class.
       if (nullMarkingForTopLevelClass == NullMarking.FULLY_UNMARKED
           && ASTHelpers.hasDirectAnnotationWithSimpleName(
               classSymbol, NullabilityUtil.NULLMARKED_SIMPLE_NAME)) {
         nullMarkingForTopLevelClass = NullMarking.PARTIALLY_MARKED;
+      } else if (nullMarkingForTopLevelClass == NullMarking.FULLY_MARKED
```

Comment:
Would this be clearer with an `||` inside a single `if`, rather than an `if` and `else if` with the same body?  I lean towards `||` but could be convinced otherwise

Reply:
Also to increase readability we could extract the condition into a separate method like `classAnnotationIntroducesPartialMarking()`